client/src/pages/Profile/ProfileEdit/ProfileEditForm/ProfileEditForm.tsx

```diff
@@ -83,10 +78,10 @@ export default function ProfileEditForm(): JSX.Element {
               email: state.email,
               gender: state.gender,
               birthday: state.birthday,
-              dobDay: state.birthday.slice(8, 10),
-              dobMonth: state.birthday.slice(5, 7),
-              dobYear: state.birthday.slice(0, 4),
-              dayArray: days(birthday.year, birthday.month),
+              dobDay: parseInt(state.birthday.slice(8, 10)) || state.dobDay,
+              dobMonth: state.birthday.slice(5, 7) || state.dobMonth,
+              dobYear: state.birthday.slice(0, 4) || state.dobYear,
```

Comment:
There are probably better Date & Time tools available in JS (`Date` objects, `moment.js` or `date-fns`) to extract/format the needed values here instead of `slice` or `substring`.